exe/CwTool.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Main where
+
+import Configuration.Utils.CommandLine
+import Options.Applicative
+import Text.PrettyPrint.ANSI.Leijen (fillSep, text, vcat)
+
+import qualified Chain2Gexf
+import qualified Ea
+import qualified RunNodes
+import qualified SlowTests
+import qualified TransactionGenerator
+
+data Command
+  = C2Gexf
+  | Ea
+  | GenTransactions
+  | RunNodes
+  | SlowTests
+
+commandParser :: Parser Command
+commandParser = subparser $
+    command "bigbang" eaOpts <> -- Using "ea" as a command seemed to have problems
+    command "chain2gexf" (C2Gexf <$ Chain2Gexf.opts) <>
+    command "gen-transactions" (GenTransactions <$ tgenOpts) <>
+    command "run-nodes" (RunNodes <$ RunNodes.runNodesOpts) <>
+    command "slow-tests" (SlowTests <$ slowTestOpts)
+
+tgenOpts :: ParserInfo Command
+tgenOpts = info (pure GenTransactions)
+    (fullDesc
+     <> progDesc "Generate a random stream of simulated blockchain transactions")
+
+eaOpts :: ParserInfo Command
+eaOpts = info (pure Ea)
+    (fullDesc
+     <> progDesc "Generate Chainweb genesis blocks and their payloads")
+
+slowTestOpts :: ParserInfo Command
+slowTestOpts = info (pure SlowTests)
+    (fullDesc
+     <> progDesc "Run slow Chainweb tests")
+
+main :: IO ()
+main = do
+    cmd <- customExecParser p opts
+    case cmd of
+      C2Gexf -> Chain2Gexf.main
+      Ea -> Ea.main
+      GenTransactions -> TransactionGenerator.main
+      RunNodes -> RunNodes.main
+      SlowTests -> SlowTests.main
```

Comment:
Will the existing option parsing still work for the sub-commands? 

Does `customExecParser` update the argument environment? 

Otherwise I would have expected something like `withArgs` that would strip the command from the options list and pass only the remaining options.

Reply:
With the exception of ea (which Colin and I are handling), I did basic testing of each one and they appear to work.   I didn't do exhaustive checking because some are more involved, but I at least ran things like `--help`.